src/hotspot/share/oops/method.cpp

```diff
@@ -423,6 +427,10 @@ void Method::remove_unshareable_info() {
 
 void Method::restore_unshareable_info(TRAPS) {
   assert(is_method() && is_valid_method(this), "ensure C++ vtable is restored");
+  if (_adapter != nullptr) {
+    _adapter->restore_unshareable_info(CHECK);
+    _from_compiled_entry = _adapter->get_c2i_entry();
```

Comment:
There's similar code in `Method::restore_archived_method_handle_intrinsic()`. I would suggest combining them into a helper function to avoid code duplication.

Reply:
Done